### DISCUSSION TITLE
How to mock a child component property value?

### DISCUSSION THREAD

**Discussion by jfpicard123** [help-me-mom/ng-mocks#5324] [Q&A] [answered]
My child component has dirty (boolean) property. How can I mock that child component and set it to return true (or false) in the dirty property? (just like we would do with services : 

`serviceMock.method.and.return(false)`

**Comment by jfpicard123** [comment]
Just figured it out .. 

```
const childComponent = spectator.query(MyChildComponent);
spyOnProperty(childComponent, 'myprop', 'get').and.returnValue(true);
```

**Reply by satanTime** [reply, marked as answer]
Hi there,

Also, you could use [`MockInstance`](https://ng-mocks.sudo.eu/api/MockInstance).

```ts
  MockInstance.scope();

  it('should render', () => {
    MockInstance(ChildComponent, 'myprop', true);

    // the rest of the test.
  });
```
